server/aesdsocket.c

```diff
@@ -1,276 +1,377 @@
+	pthread_mutex_lock(&mutex);
+	lseek(file_fd, 0, SEEK_SET);
+	char send_buffer[total_packet_bytes]; 			 
+	if(read(file_fd,&send_buffer,total_packet_bytes)==-1) 
+	{  
+		syslog(LOG_ERR, "Cannot read the file");
+		return -1;
+	}	
+	pthread_mutex_unlock(&mutex); 
 	
-	freeaddrinfo(servinfo); //Free this memory before next operations
-	
-	fd_status=open("/var/tmp/aesdsocketdata.txt", O_WRONLY | O_CREAT | O_TRUNC, S_IRWXU | S_IRWXG | S_IRWXO); //To create a file and give permissions to all
-	if (fd_status==-1)
-	{
-		syslog(LOG_ERR, "The file could not be created/found");
-		exit(7);
-	}
-	close(fd_status);
-	
-	//Infinite loop for server-client connections
-	while(1)
+	//Send data packet to the client 
+	if(send(accept_connection,&send_buffer,total_packet_bytes,0)==-1) 
 	{
-	
-	//Listen
-	listen_status=listen(socket_fd,20); //Max queue of connections set to 20
-	if(listen_status==-1)
+		syslog(LOG_ERR, "Unable to send the bytes to the socket");
+		return -1;
+	} 
+	return 0;
+}
+
+//Threads packet transfers
+void* thread_packet_data(void* thread_in_action) 
+{
+	char *temp_buff = NULL; 
+	int bytes_per_packet =1;
+	int recv_status =0,new_line =0;
+	int conn_close =0;
+
+	if(thread_in_action == NULL)
+  		exit(7);
+
+	//logic to receive packets from client
+	thread_nodes_t *t_node_params= (thread_nodes_t *)thread_in_action;
+	temp_buff = (char*)malloc(sizeof(char));
+	if (temp_buff == NULL) 
 	{
-		syslog(LOG_ERR, "Listening to the connections failed");
+		syslog(LOG_ERR,"Couldn't allocate memory to store packets");
+		t_node_params->thread_complete = true;
+		close(t_node_params->accept_connection); 
 		exit(8);
-	}
-	
-	//accept
-	accept_return=accept(socket_fd,(struct sockaddr *)&client_addr,&size);
-	if(accept_return==-1)
-	{
-		syslog(LOG_ERR, "Connection could not be accepted");
-		exit(9);
-	}
-	else
-	{
-		syslog(LOG_INFO,"Accepts connection from %s",inet_ntoa(client_addr.sin_addr));
-		printf("Accepts connection from %s\n",inet_ntoa(client_addr.sin_addr));
-	}
-	
-	
-	store_data = (char*)malloc(100); //Allocate a memory
-	if(store_data==NULL)
-	{
-		syslog(LOG_ERR, "Memory couldn't be allocated");
-		exit(10);
-	}
-	
-	memset(store_data,0,100); //Reset the temporary buffer
-	
-	write_flag=false;
-	
-	//Loop to receive data packet
-	while(!write_flag)
-	{
-	
-	rec_status=recv(accept_return,buff,100,0); //Receive data packets from client
-	if(rec_status==-1)
-	{
-		syslog(LOG_ERR, "Error in reception of data packets from client");
-		exit(11);
-	}
-	
-	
-	for(i=0;i<100;i++)
+	}	
+	while (!conn_close && !handler_status)
 	{
-		if(buff[i]=='\n')
+		new_line = 0;
+		while((!new_line) && (!conn_close) && (!handler_status))
 		{
-			i++;
-			write_flag=true;
-			break;
-		}	
+			recv_status = recv(t_node_params -> accept_connection,temp_buff+bytes_per_packet-1,1,0);
+			if (recv_status == -1)
+			{
+				syslog(LOG_ERR,"Cannot receive bytes");
+				t_node_params->thread_complete= true;
+				close(t_node_params->accept_connection); 
+				exit(9);
```

Comment:
@shashankchandrasekaran is killing the entire process teh correct way to handle errors?  What do the requirements say about handling errors in your thread?